partnership/partnership.go

```diff
@@ -36,25 +43,159 @@ func Partnership(w http.ResponseWriter, r *http.Request) {
 }
 
 func LifePartnershipFx(resp http.ResponseWriter, r *http.Request) (string, interface{}, error) {
-	var (
-		policy   models.Policy
-		person   models.User
-		asset    models.Asset
-		response PartnershipResponse
-	)
+	var response PartnershipResponse
 	resp.Header().Set("Access-Control-Allow-Methods", "GET")
+	resp.Header().Set("Content-Type", "application/json")
+
+	log.Println("[LifePartnershipFx] handler start ----------------------------")
 
+	partnershipUid := strings.ToLower(r.Header.Get("partnershipUid"))
 	jwtData := r.URL.Query().Get("jwt")
 
-	// decode JWT
-	// Parse takes the token string and a function for looking up the key. The latter is especially
-	// useful if you use multiple keys for your application.  The standard is to use 'kid' in the
-	// head of the token to identify which key to use, but the parsed token (head and claims) is provided
-	// to the callback, providing flexibility.
+	log.Printf("[LifePartnershipFx] partnershipUid: %s jwt: %s", partnershipUid, jwtData)
+
+	policy, product, node, err := LifePartnership(partnershipUid, jwtData, r.Header.Get("Origin"))
+
+	response.Policy = policy
+	response.Product = product
+	response.Partnership = *node.Partnership
+
+	responseJson, err := json.Marshal(response)
+
+	log.Printf("[LifePartnershipFx] response: %s", string(responseJson))
+
+	return string(responseJson), response, err
+}
+
+func LifePartnership(partnershipUid, jwtData, origin string) (models.Policy, models.Product, *models.NetworkNode, error) {
+	var (
+		policy          models.Policy
+		productLife     models.Product
+		partnershipNode *models.NetworkNode
+		err error
+	)
+
+	log.Printf("[LifePartnership]")
+
+	if partnershipNode, err = network.GetNodeByUid(partnershipUid); err != nil {
+		return policy, productLife, partnershipNode, err
+	}
+
+	partnershipName := partnershipNode.Partnership.Name
+
+	log.Printf("[LifePartnership] loading latest life product")
+
+	products := partnershipNode.Products
+	productLife = getLatestLifeProduct(products)
+
+	log.Printf("[LifePartnership] loading life product by channel")
+
+	ecommerceProducts := product.GetAllProductsByChannel(models.ECommerceChannel)
+	latestLifeProduct := getLatestLifeProduct(ecommerceProducts)
+
+	log.Printf("[LifePartnership] setting policy basic info")
+
+	policy.Name = productLife.Name
+	policy.NameDesc = *productLife.NameDesc
+	policy.ProductVersion = productLife.Version
+	policy.Company = productLife.Companies[0].Name
+	policy.ProducerUid = partnershipUid
+	policy.ProducerCode = partnershipName
+	policy.PartnershipName = partnershipName
+	policy.ProducerType = partnershipNode.Type
+
+	switch partnershipName {
+	case models.PartnershipBeProf:
+		log.Println("[LifePartnership] call beProfPartnership function")
+		err = beProfPartnership(jwtData, &policy, &latestLifeProduct)
+	case models.PartnershipFacile:
+		log.Println("[LifePartnership] call facilePartnership function")
+		err = facilePartnership(jwtData, &policy, &latestLifeProduct)
+	}
```

Comment:
I'd put a default that returns an error, otherwise the code continues with an unmodified policy